pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+const supportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	newVersionMajor := newVersion.Major()
+	newVersionMinor := newVersion.Minor()
+	oldVersionMajor := oldVersion.Major()
+	oldVersionMinor := oldVersion.Minor()
+
+	majorVersionDifference := math.Abs(float64(newVersionMajor) - float64(oldVersionMajor))
+
+	if majorVersionDifference > 0 {
+		return fmt.Errorf("MAJOR version upgrades are not supported. Major version difference between upgrade version (%d.%d) and server version (%d.%d) is %f", newVersionMajor, newVersionMinor, oldVersionMajor, oldVersionMinor, majorVersionDifference)
+	}
+
+	minorVersionDifference := float64(newVersionMinor) - float64(oldVersionMinor)
+
+	if minorVersionDifference < 0 {
+		return fmt.Errorf("MINOR version downgrade is not supported. Difference between upgrade version (%d.%d) and server version (%d.%d) should meet the supported version increment of +%d", newVersionMajor, newVersionMinor, oldVersionMajor, oldVersionMinor, supportedMinorVersionIncrement)
```

Comment:
nit: this error message sounds very programmer oriented
Can it be simplified to something like "blablabla only %d minor version skew is supported blabla"

Reply:
sure, I can make it simpler.